internal/components/sdd/vscode_models.go

```
@@ -0,0 +1,268 @@
+		path = opencode.DefaultCachePath()
+	}
+	if path == "" {
+		return nil, fmt.Errorf("models cache path unavailable")
+	}
+
+	providers, err := opencode.LoadModels(path)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			return nil, fmt.Errorf("models cache %q not found", path)
+		}
+		return nil, fmt.Errorf("read models cache %q: %v", path, err)
```

golangci-lint (2.12.2)

[error] non-wrapping format verb for fmt.Errorf. Use `%w` to format errors

(errorlint)